x/data/simulation/operations.go

```diff
@@ -0,0 +1,378 @@
+func WeightedOperations(
+	appParams simtypes.AppParams, cdc codec.JSONCodec,
+	ak data.AccountKeeper, bk data.BankKeeper,
+	qryClient data.QueryClient) simulation.WeightedOperations {
+
+	var (
+		weightMsgAnchor           int
+		weightMsgAttest           int
+		weightMsgDefineResolver   int
+		weightMsgRegisterResolver int
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgAnchor, &weightMsgAnchor, nil,
+		func(_ *rand.Rand) {
+			weightMsgAnchor = WeightAnchor
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgAttest, &weightMsgAttest, nil,
+		func(_ *rand.Rand) {
+			weightMsgAttest = WeightAttest
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgDefineResolver, &weightMsgDefineResolver, nil,
+		func(_ *rand.Rand) {
+			weightMsgDefineResolver = WeightDefineResolver
+		},
+	)
+
+	appParams.GetOrGenerate(cdc, OpWeightMsgRegisterResolver, &weightMsgRegisterResolver, nil,
+		func(_ *rand.Rand) {
+			weightMsgRegisterResolver = WeightRegisterResolver
+		},
+	)
+
+	ops := simulation.WeightedOperations{
+		simulation.NewWeightedOperation(
+			weightMsgAnchor,
+			SimulateMsgAnchor(ak, bk),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgAnchor,
+			SimulateMsgAttest(ak, bk),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgAnchor,
+			SimulateMsgDefineResolver(ak, bk),
+		),
+		simulation.NewWeightedOperation(
+			weightMsgAnchor,
+			SimulateMsgRegisterResolver(ak, bk, qryClient),
+		),
+	}
+
+	return ops
+}
+
+// SimulateMsgAnchor generates a MsgAnchor with random values.
+func SimulateMsgAnchor(ak data.AccountKeeper, bk data.BankKeeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, sdkCtx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		sender, _ := simtypes.RandomAcc(r, accs)
+
+		contentHash, err := getContentHash(r)
+		if err != nil {
+			return simtypes.NoOpMsg(ModuleName, TypeMsgAnchor, err.Error()), nil, err
+		}
+
+		spendable := bk.SpendableCoins(sdkCtx, sender.Address)
+		fees, err := simtypes.RandomFees(r, sdkCtx, spendable)
+		if err != nil {
+			return simtypes.NoOpMsg(ModuleName, TypeMsgAnchor, "fee error"), nil, err
+		}
+
+		msg := &data.MsgAnchor{
+			Sender: sender.Address.String(),
+			Hash:   contentHash,
+		}
+
+		account := ak.GetAccount(sdkCtx, sender.Address)
+		txGen := simappparams.MakeTestEncodingConfig().TxConfig
+		tx, err := helpers.GenTx(
+			txGen,
+			[]sdk.Msg{msg},
+			fees,
+			helpers.DefaultGenTxGas,
+			chainID,
+			[]uint64{account.GetAccountNumber()},
+			[]uint64{account.GetSequence()},
+			sender.PrivKey,
+		)
+		if err != nil {
+			return simtypes.NoOpMsg(ModuleName, TypeMsgAnchor, "unable to generate mock tx"), nil, err
+		}
+
+		_, _, err = app.Deliver(txGen.TxEncoder(), tx)
+		if err != nil {
+			return simtypes.NoOpMsg(ModuleName, TypeMsgAnchor, "unable to deliver tx"), nil, err
+		}
```

Comment:
lines 127-141 are "almost" repeated in other functions below. Let's create a new helper function to generate and deliver a transaction